lib/preview/previewUtils.js

```diff
@@ -95,14 +97,60 @@ const silenceConsoleWhile = async (act, ...args) => {
   return result;
 }
 
+const memoize = (func, cacheBustCallback) => {
+  const cache = {};
+  return async (...args) => {
+    const stringArgs = args.toString();
+    const storedResult = cache[stringArgs];
+    if (storedResult && (cacheBustCallback ? !cacheBustCallback() : true)) {
+      //console.log(`Cache hit ${func}`)
+      return storedResult;
+    }
+    //console.log(`Cache miss ${func}`)
+    const res = await func(...args);
+    cache[stringArgs] = res;
+    return res;
+  }
+}
```

Comment:
Very simple caching mechanism to avoid repeating those metadata fetches unnecessarily - `cacheBustCallback` here is optional and not implemented for any usages yet. 
Since the initial fetch for templates/modules that the index page does also returns `previewKeys` for said assets, we may want to try and prepopulate the cache with those, in which case this caching model doesn't work as well - a little more brainstorming needed I think on how best to do this.